src/GovUk.Education.ExploreEducationStatistics.Public.Data.Api/Requests/Converters/DataSetQueryLocationJsonConverter.cs

```diff
@@ -0,0 +1,120 @@
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using GovUk.Education.ExploreEducationStatistics.Common.Extensions;
+using GovUk.Education.ExploreEducationStatistics.Common.Model.Data;
+using GovUk.Education.ExploreEducationStatistics.Common.Utils;
+
+namespace GovUk.Education.ExploreEducationStatistics.Public.Data.Api.Requests.Converters;
+
+public class DataSetQueryLocationJsonConverter : JsonConverter<DataSetQueryLocation>
+{
+    public override bool CanConvert(Type type)
+    {
+        return type.IsAssignableFrom(typeof(DataSetQueryLocation));
+    }
+
+    public override DataSetQueryLocation? Read(
+        ref Utf8JsonReader reader,
+        Type typeToConvert,
+        JsonSerializerOptions options)
+    {
+        if (!JsonDocument.TryParseValue(ref reader, out var doc))
+        {
+            throw new JsonException();
+        }
+
+        var rootElement = doc.RootElement.GetRawText();
+
+        var propertyNames = doc.RootElement
+            .EnumerateObject()
+            .Select(p => p.Name.ToUpperFirst())
+            .ToHashSet();
+
+        if (!doc.RootElement.TryGetProperty(nameof(DataSetQueryLocation.Level).ToLowerFirst(), out var levelProperty))
+        {
+            throw new JsonException();
+        }
+
+        try
+        {
+            var levelString = levelProperty.GetString() ?? string.Empty;
```

Comment:
No we don't want to throw. If possible, we want to avoid throwing so that we can actually deserialize to a request model to allow the validation layer to hit.

Not being able to deserialize usually leads to a worse user experience as it prevents us from providing more specific errors to the user.